modules/cache-common/src/main/java/org/opensearch/cache/common/tier/TieredSpilloverCache.java

```diff
@@ -159,6 +180,10 @@ public V computeIfAbsent(ICacheKey<K> key, LoadAwareCacheLoader<ICacheKey<K>, V>
             V value = null;
             try (ReleasableLock ignore = writeLock.acquire()) {
                 value = onHeapCache.computeIfAbsent(key, loader);
+                if (loader.isLoaded()) {
+                    // The value was just computed and added to the cache
+                    updateStatsOnPut(TIER_DIMENSION_VALUE_ON_HEAP, key, value);
+                }
```

Comment:
We might have to handle more scenarios here :
1. In case there are multiple concurrent requests for the same key, `getValueFromTieredCache()` will return null for all. But during computeIfAbsent(), only one will load the value and rest of them will essentially have `hits`.
```
if (loader.isLoaded()) {
     // miss
} else {
     // hit
}
```
2. Following up to above point, I see you are already capturing misses/hits in `getValueFromTieredCache()` which may be wrong. As it may happen like in Scenario#1 above, a value for a key maybe null in `get()` but we may still end up finding it in `computeIfAbsent`. You might have to change `getValueFromTieredCache()` to `getValueFromTieredCache(boolean captureStats)` or something like that.

A unit test for this would also help.

Reply:
Also can we move stats related logic outside write lock? As I don't want a thread holding locks for that, can be expensive for reads.